lxd/device/nic_physical.go

```diff
@@ -29,15 +32,60 @@ func (d *nicPhysical) validateConfig(instConf instance.ConfigReader) error {
 		return ErrUnsupportedDevType
 	}
 
-	requiredFields := []string{"parent"}
+	requiredFields := []string{}
 	optionalFields := []string{
+		"parent",
 		"name",
 		"maas.subnet.ipv4",
 		"maas.subnet.ipv6",
 		"boot.priority",
 		"gvrp",
 	}
 
+	if d.config["network"] != "" {
+		requiredFields = append(requiredFields, "network")
+
+		bannedKeys := []string{"nictype", "parent", "mtu", "vlan", "maas.subnet.ipv4", "maas.subnet.ipv6", "gvrp"}
+		for _, bannedKey := range bannedKeys {
+			if d.config[bannedKey] != "" {
+				return fmt.Errorf("Cannot use %q property in conjunction with %q property", bannedKey, "network")
+			}
+		}
+
+		// If network property is specified, lookup network settings and apply them to the device's config.
+		// project.Default is used here as physical networks don't support projects.
+		var err error
+		d.network, err = network.LoadByName(d.state, project.Default, d.config["network"])
+		if err != nil {
+			return fmt.Errorf("Error loading network config for %q: %w", d.config["network"], err)
+		}
+
+		if d.network.Status() != api.NetworkStatusCreated {
+			return fmt.Errorf("Specified network is not fully created")
+		}
+
+		if d.network.Type() != "physical" {
+			return fmt.Errorf("Specified network must be of type physical")
+		}
+
+		netConfig := d.network.Config()
+
+		// Get actual parent device from network's parent setting.
+		d.config["parent"] = netConfig["parent"]
+
+		// Copy certain keys verbatim from the network's settings.
+		inheritKeys := []string{"mtu", "vlan", "maas.subnet.ipv4", "maas.subnet.ipv6", "gvrp"}
+		for _, inheritKey := range inheritKeys {
+			_, found := netConfig[inheritKey]
+			if found {
+				d.config[inheritKey] = netConfig[inheritKey]
+			}
+		}
+	} else {
+		// If no network property supplied, then parent property is required.
+		requiredFields = append(requiredFields, "parent")
+	}
+
 	if instConf.Type() == instancetype.Container || instConf.Type() == instancetype.Any {
```

Comment:
@gabrielmougard So I think this will be an issue for VMs.

If the physical network being used has one of the fields below `"mtu", "hwaddr", "vlan"` then they will be added to the device's config on line 77, and then will fail validation here, because the instance type will be `instancetype.VM` and thus the field will not be considered optional.

I think instead we should move this stanza above the new stanza you've added in this PR, and then in the `inheritKeys` section, only inherit the key from the network if its in the `optionalFields` list.

Reply:
Alright, changing that